server/src/main/java/org/elasticsearch/repositories/RepositoriesStats.java

```diff
@@ -0,0 +1,112 @@
+
+    public RepositoriesStats(StreamInput in) throws IOException {
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_010)) {
+            repositoryThrottlingStats = in.readMap(StreamInput::readString, ThrottlingStats::new);
+        } else {
+            repositoryThrottlingStats = new HashMap<>();
+        }
+    }
+
+    public RepositoriesStats(Map<String, ThrottlingStats> repositoryThrottlingStats) {
+        this.repositoryThrottlingStats = new HashMap<>(repositoryThrottlingStats);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_010)) {
+            out.writeMap(repositoryThrottlingStats, StreamOutput::writeString, (o, v) -> v.writeTo(o));
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field("repository", repositoryThrottlingStats);
+        return builder;
+    }
+
+    public Map<String, ThrottlingStats> getRepositoryThrottlingStats() {
+        return Collections.unmodifiableMap(repositoryThrottlingStats);
+    }
+
+    public static class Builder {
+        private final Map<String, ThrottlingStats> repoThrottlingStats = new HashMap<>();
+
+        public Builder add(String repoName, long totalReadThrottledNanos, long totalWriteThrottledNanos) {
+            repoThrottlingStats.put(repoName, new ThrottlingStats(totalReadThrottledNanos, totalWriteThrottledNanos));
+            return this;
+        }
+
+        public RepositoriesStats build() {
+            return new RepositoriesStats(repoThrottlingStats);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class ThrottlingStats implements ToXContentObject, Writeable {
+
+        private final long totalReadThrottledNanos;
+        private final long totalWriteThrottledNanos;
+
+        ThrottlingStats(StreamInput in) throws IOException {
+            this.totalReadThrottledNanos = in.readLong();
+            this.totalWriteThrottledNanos = in.readLong();
+        }
+
+        public ThrottlingStats(long totalReadThrottledNanos, long totalWriteThrottledNanos) {
+            this.totalReadThrottledNanos = totalReadThrottledNanos;
+            this.totalWriteThrottledNanos = totalWriteThrottledNanos;
+        }
+
+        public long getTotalReadThrottledNanos() {
+            return totalReadThrottledNanos;
+        }
+
+        public long getTotalWriteThrottledNanos() {
+            return totalWriteThrottledNanos;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field("total_read_throttled_nanos", totalReadThrottledNanos);
+            builder.field("total_write_throttled_nanos", totalWriteThrottledNanos);
```

Comment:
Could we also have these fields represented in a human-readable fashion if the `?human` flag is specified? IIRC that's a little awkward with nanoseconds, because if you use `org.elasticsearch.xcontent.XContentBuilder#humanReadableField` with a `TimeValue` then you get the raw output in milliseconds (because of [this](https://github.com/elastic/elasticsearch/blob/12ad399c488f0cc60e19b5e1b29c6d569cb4351a/server/src/main/java/org/elasticsearch/common/xcontent/XContentElasticsearchExtension.java#L96)) but still we can just check the human flag directly (like we do e.g. [here](https://github.com/elastic/elasticsearch/blob/12ad399c488f0cc60e19b5e1b29c6d569cb4351a/server/src/main/java/org/elasticsearch/node/AdaptiveSelectionStats.java#L66-L69)).

Reply:
Those values are just *long* counters (not wrapped in TimeValue) and I think those transformers won't affect them, right?

Reply:
That's correct.

Reply:
I think this is the only remaining point now.

Reply:
Following ...

1) http://localhost:9200/_nodes/rVt20tYyS3yN5LNYHjPoSw/stats/repository?pretty
```
{
  "_nodes" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  },
  "cluster_name" : "stateless",
  "nodes" : {
    "rVt20tYyS3yN5LNYHjPoSw" : {
      "timestamp" : 1686604771491,
      "name" : "es01",
      "transport_address" : "172.18.0.4:9300",
      "host" : "172.18.0.4",
      "ip" : "172.18.0.4:9300",
      "roles" : [
        "index",
        "master"
      ],
      "attributes" : {
        "stateless.memory" : "8232894464",
        "xpack.installed" : "true"
      },
      "repositories" : {
        "my_repository" : {
          "total_read_throttled_time_nanos" : 0,
          "total_write_throttled_time_nanos" : 551110710
        }
      }
    }
  }
}
```

2) http://localhost:9200/_nodes/rVt20tYyS3yN5LNYHjPoSw/stats/repository?pretty&human
```
{
  "_nodes" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  },
  "cluster_name" : "stateless",
  "nodes" : {
    "rVt20tYyS3yN5LNYHjPoSw" : {
      "timestamp" : 1686604850299,
      "name" : "es01",
      "transport_address" : "172.18.0.4:9300",
      "host" : "172.18.0.4",
      "ip" : "172.18.0.4:9300",
      "roles" : [
        "index",
        "master"
      ],
      "attributes" : {
        "stateless.memory" : "8232894464",
        "xpack.installed" : "true"
      },
      "repositories" : {
        "my_repository" : {
          "total_read_throttled_time" : "0s",
          "total_write_throttled_time" : "551.1ms"
        }
      }
    }
  }
}
```

Note that field names are different for human and non-human mode. asciidoc was updated.

Reply:
I think this needs to follow the same pattern/convention as in [AdaptiveSelectionStats](https://github.com/elastic/elasticsearch/blob/12ad399c488f0cc60e19b5e1b29c6d569cb4351a/server/src/main/java/org/elasticsearch/node/AdaptiveSelectionStats.java#L66-L69): we always output the machine-readable field, _plus_ the human-readable field if requested.